python/ray/experimental/channel/shared_memory_channel.py

```diff
@@ -480,6 +484,71 @@ def close(self) -> None:
         self._worker.core_worker.experimental_channel_set_error(self._reader_ref)
 
 
+@DeveloperAPI
+class BufferedSharedMemoryChannel(ChannelInterface):
+    """A Channel that allow to buffer inputs.
+
+    The write/read APIs are non-blocking as long as the buffers are not
+    exhausted. The caller of the API should guarantee to consume buffers
+    before they are exhausted, otherwise write/read raises
+    `RayChannelTimeoutError`.
```

Comment:
Maybe saying something like the following is more accurate:
If the buffer is full for writes or empty for reads and the operation blocks longer than configured timeouts, `RayChannelTimeoutError` will be raised.

Reply:
updated accordingly.